lib/go/thrift/application_exception.go

```diff
@@ -59,9 +61,119 @@ type TApplicationException interface {
 	Write(ctx context.Context, oprot TProtocol) error
 }
 
-type tApplicationException struct {
+type ValidationError struct {
 	message string
-	type_   int32
+	check   string
+}
+
+func (e *ValidationError) Check() string {
+	return e.check
+}
+
+func (e ValidationError) Error() string {
+	return e.message
+}
+
+func (p *ValidationError) Read(ctx context.Context, iprot TProtocol) error {
+	_, err := iprot.ReadStructBegin(ctx)
+	if err != nil {
+		return err
+	}
+
+	message := ""
+	check := ""
+
+	for {
+		_, ttype, id, err := iprot.ReadFieldBegin(ctx)
+		if err != nil {
+			return err
+		}
+		if ttype == STOP {
+			break
+		}
+		switch id {
+		case 1:
+			if ttype == STRING {
+				if message, err = iprot.ReadString(ctx); err != nil {
+					return err
+				}
+			} else {
+				if err = SkipDefaultDepth(ctx, iprot, ttype); err != nil {
+					return err
+				}
+			}
+		case 2:
+			if ttype == STRING {
+				if check, err = iprot.ReadString(ctx); err != nil {
+					return err
+				}
+			} else {
+				if err = SkipDefaultDepth(ctx, iprot, ttype); err != nil {
+					return err
+				}
+			}
+		default:
+			if err = SkipDefaultDepth(ctx, iprot, ttype); err != nil {
+				return err
+			}
+		}
+		if err = iprot.ReadFieldEnd(ctx); err != nil {
+			return err
+		}
+	}
+	if err := iprot.ReadStructEnd(ctx); err != nil {
+		return err
+	}
+
+	p.message = message
+	p.check = check
+
+	return nil
+}
+
+func (p *ValidationError) Write(ctx context.Context, oprot TProtocol) (err error) {
+	err = oprot.WriteStructBegin(ctx, "ValidationError")
+	if err != nil {
+		return
+	}
+	if len(p.Error()) > 0 {
+		err = oprot.WriteFieldBegin(ctx, "message", STRING, 1)
+		if err != nil {
+			return
+		}
+		err = oprot.WriteString(ctx, p.message)
+		if err != nil {
+			return
+		}
+		err = oprot.WriteFieldEnd(ctx)
+		if err != nil {
+			return
+		}
+		err = oprot.WriteFieldBegin(ctx, "check", STRING, 2)
+		if err != nil {
+			return
+		}
+		err = oprot.WriteString(ctx, p.check)
+		if err != nil {
+			return
+		}
+		err = oprot.WriteFieldEnd(ctx)
+		if err != nil {
+			return
+		}
+	}
+	err = oprot.WriteFieldStop(ctx)
+	if err != nil {
+		return
+	}
+	err = oprot.WriteStructEnd(ctx)
+	return
+}
+
+type tApplicationException struct {
+	message         string
+	type_           int32
+	validationError *ValidationError
```

Comment:
this should not be specified to `validationError *ValidationError`, but just a generic `err error`. see `TTransportException` for an example: https://github.com/apache/thrift/blob/e0ee2c7514b812b5a18bfe9aab9594c5879dc34f/lib/go/thrift/transport_exception.go#L48